### ISSUE TITLE
[BUG] SUEntityGetAttributeDictionary() in Live model

### ISSUE THREAD

**Issue by tommash** [SketchUp/api-issue-tracker#318]
SU Version: Any with C SDK API

SUEntityGetAttributeDictionary() creates a dictionary, also in a live model, which in turn destroys dispatching of observer events!

SUEntityGetAttributeDictionary() should never create the dictionary! There is a special function for this.

Even Ruby function attribute_dictionary(name, create = false), has the creation disabled by default.

**Comment by DanRathbun** [comment]
Yes the default should match the Ruby API, and there should have been a dedicated boolean parameter for creation right from the start.

(You just know Thomas will ask for a code snippet to reproduce the error.)

**Comment by DanRathbun** [comment]
I suppose you have to iterate the attribute dictionaries collection getting each's name and compare with a passed string ?

**Comment by tommash** [comment]
Correct. I got trough all the dictionaries for the given entity, find the one I want to read and read a value from the dictionary.
`# encoding: UTF-8

require "sketchup.rb"

module NONAME

	# Selection observer

	class MySelObserver < Sketchup::SelectionObserver
		def onSelectionBulkChange(selection)
			puts "Selection Bulk Change"
		end

		def onSelectionCleared(selection)
			puts "Selection Cleared"
		end
	end

	## Watch if current view changed
	class MyViewObs < Sketchup::ViewObserver

		def onViewChanged(view)
			puts "View changed"
		end
	end

	class MyAppObs < Sketchup::AppObserver
		def onNewModel(model)
			model.active_view.add_observer NONAME::MyViewObs.new
			model.selection.add_observer NONAME::MySelObserver.new
		end

		def onActivateModel(model)
			model.active_view.add_observer NONAME::MyViewObs.new
			model.selection.add_observer NONAME::MySelObserver.new
		end

		def onOpenModel(model)
			model.active_view.add_observer NONAME::MyViewObs.new
			model.selection.add_observer NONAME::MySelObserver.new
		end

		def expectsStartupModelNotifications
			return true
		end
	end

	Sketchup.add_observer MyAppObs.new

end
`

**Comment by DanRathbun** [comment]
You need to use:

\`\`\`ruby
  \# code here
\`\`\`

Single backtick only works for text on the same line.

**Comment by taustin73** [comment]
SKOR-11818
